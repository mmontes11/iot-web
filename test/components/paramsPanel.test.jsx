import React from "react";
import ParamsPanel from "components/paramsPanel";
import { shallow } from "enzyme";

describe("components/paramsPanel", () => {
  it("renders in normal state", () => {
    const wrapper = shallow(
      <ParamsPanel
        type={{
          label: "Select type",
          items: ["foo", "bar"],
          isActive: false,
          onButtonClick: () => undefined,
          onItemClick: () => undefined,
        }}
        observation={{
          label: "Select type",
          items: ["foo", "bar"],
          isActive: false,
          isLoading: false,
          isDisabled: false,
          onButtonClick: () => undefined,
          onItemClick: () => undefined,
        }}
        reset={{
          isDisabled: true,
          onReset: () => undefined,
        }}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it("renders after some params have been specified", () => {
    const wrapper = shallow(
      <ParamsPanel
        type={{
          label: "Select type",
          items: ["foo", "bar"],
          isActive: false,
          onButtonClick: () => undefined,
          onItemClick: () => undefined,
        }}
        observation={{
          label: "Select type",
          items: ["foo", "bar"],
          isActive: false,
          isLoading: false,
          isDisabled: false,
          onButtonClick: () => undefined,
          onItemClick: () => undefined,
        }}
        reset={{
          isDisabled: false,
          onReset: () => undefined,
        }}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it("simulates click on reset", () => {
    const onReset = jest.fn();
    const wrapper = shallow(
      <ParamsPanel
        type={{
          label: "Select type",
          items: ["foo", "bar"],
          isActive: false,
          onButtonClick: () => undefined,
          onItemClick: () => undefined,
        }}
        observation={{
          label: "Select type",
          items: ["foo", "bar"],
          isActive: false,
          isLoading: false,
          isDisabled: false,
          onButtonClick: () => undefined,
          onItemClick: () => undefined,
        }}
        reset={{
          isDisabled: false,
          onReset,
        }}
      />,
    );
    wrapper.find(".button").simulate("click");
    expect(onReset).toHaveBeenCalledTimes(1);
  });
});
