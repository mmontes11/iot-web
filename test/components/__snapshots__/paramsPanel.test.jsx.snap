// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/paramsPanel renders after some params have been specified 1`] = `
<div
  className="box"
>
  <div
    className="columns"
  >
    <div
      className="column is-5"
    >
      <Dropdown
        isActive={false}
        isDisabled={false}
        isLoading={false}
        items={
          Array [
            "foo",
            "bar",
          ]
        }
        label="Select type"
        onButtonClick={[Function]}
        onItemClick={[Function]}
      />
    </div>
    <div
      className="column is-5"
    >
      <Dropdown
        isActive={false}
        isDisabled={false}
        isLoading={false}
        items={
          Array [
            "foo",
            "bar",
          ]
        }
        label="Select type"
        onButtonClick={[Function]}
        onItemClick={[Function]}
      />
    </div>
    <div
      className="column"
    >
      <button
        className="button is-warning is-fullwidth"
        disabled={false}
        onClick={[Function]}
      >
        <span>
          Reset
        </span>
        <span
          className="icon is-small"
        >
          <i
            aria-hidden="true"
            className="fas fa-eraser"
          />
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`components/paramsPanel renders in normal state 1`] = `
<div
  className="box"
>
  <div
    className="columns"
  >
    <div
      className="column is-5"
    >
      <Dropdown
        isActive={false}
        isDisabled={false}
        isLoading={false}
        items={
          Array [
            "foo",
            "bar",
          ]
        }
        label="Select type"
        onButtonClick={[Function]}
        onItemClick={[Function]}
      />
    </div>
    <div
      className="column is-5"
    >
      <Dropdown
        isActive={false}
        isDisabled={false}
        isLoading={false}
        items={
          Array [
            "foo",
            "bar",
          ]
        }
        label="Select type"
        onButtonClick={[Function]}
        onItemClick={[Function]}
      />
    </div>
    <div
      className="column"
    >
      <button
        className="button is-warning is-fullwidth"
        disabled={true}
        onClick={[Function]}
      >
        <span>
          Reset
        </span>
        <span
          className="icon is-small"
        >
          <i
            aria-hidden="true"
            className="fas fa-eraser"
          />
        </span>
      </button>
    </div>
  </div>
</div>
`;
